src/browser_action/on-each-interaction.js

```diff
@@ -11,38 +11,133 @@
  limitations under the License.
 */
 
+
+let recentLoAFs = [];
+
+const getIntersectingLoAFs = (start, end) => {
+  const intersectingLoAFs = [];
+
+  for (let i = 0, loaf; (loaf = recentLoAFs[i]); i++) {
+    // If the LoAF ends before the given start time, ignore it.
+    if (loaf.startTime + loaf.duration < start) continue;
+
+    // If the LoAF starts after the given end time, ignore it and all
+    // subsequent pending LoAFs (because they're in time order).
+    if (loaf.startTime > end) break;
+
+    // Still here? If so this LoAF intersects with the interaction.
+    intersectingLoAFs.push(loaf);
+  }
+  return intersectingLoAFs;
+};
+
+const loafObserver = new PerformanceObserver((list) => {
+
+  for (const entry of list.getEntries()) { recentLoAFs.push(entry) };
+  // We report interactions immediately, so don't need to keep many LoAFs around.
+  // Let's keep the last 5.
+  recentLoAFs = recentLoAFs.slice(-5);
+
+});
+loafObserver.observe({
```

Comment:
IIUC this means the LoAF observer is running unconditionally, regardless of whether the user has enabled logging, right?

Why not initialize this in the same place as the `event` observer?

Reply:
Fixed!